src/persist-client/src/critical.rs

```diff
@@ -295,6 +297,32 @@ where
         }
     }
 
+    /// Returns aggregate statistics about the contents of the shard TVC at the
+    /// given frontier.
```

Comment:
What happens if the requested `as_of` is between the lower and upper of a particular batch? It's not clear to me how we could provide these counts at anything but a batch boundary, without actually fetching and filtering the data itself...

Reply:
the num_updates on the returned struct is documented to be an overestimate

Reply:
To unpack a bit: the stats we return are the stats as of the end of the batch that "contains" the given `as_of`, and the total number of updates is an upper bound on... I'm having trouble coming up with a crisp way to put this, but something like "the number of updates that persist_source would emit if you snapshot the source at the given `as_of`".

I've been getting myself a bit hung up on the philosophical question about what exactly are the "contents of a shard", especially since the relevant set of updates may change as the seqno grows and the shard compacts... but in any case what we're capturing here does seem to be the useful notion for a cost model!

Reply:
> the number of updates that persist_source would emit if you snapshot the source at the given as_of

I also can't come up with anything better here. added this verbatim to the rustdoc on SnapshotStats

> "contents of a shard", especially since the relevant set of updates may change as the seqno grows and the shard compacts

oooo, fantastic point! compaction can only make the number of updates go down, so what we're doing is still an upper bound. called this out in the rustdoc, too (this one more for us than for compute)